include/nanogui/widget.h

```diff
@@ -334,7 +413,66 @@ class NANOGUI_EXPORT Widget : public Object {
      * \endrst
      */
     float mIconExtraScale;
+
     Cursor mCursor;
+
+private:
+    /* Font related utilities.  To use in derived classes, call the getter.
+     * E.g. for `mFont`, call `font()` and `mTooltipFont` call `tooltipFont()`.
+     * To set it, call the setter.  E.g. set `mFont` with `setFont` and
+     * `mTooltipFont` with `setTooltipFont`.
+     */
+
+    /**
+     * When \ref setTheme is called, this method configures both \ref mFont and
+     * \ref mTooltipFont.
+     */
+    void setDefaultFonts();
+
+    /**
+     * Used to set the font size of a widget explicitly.  The initial value is
+     * ``-1``, and a negative number indicates that the theme's font size should
+     * be used instead.  Specifically, \ref fontSize will return
+     * \ref Theme::mStandardFontSize when ``mFontSize < 0``.
+     */
+    float mFontSize;
+
+    /// The current font face being used to draw text (if this Widget displays text).
+    std::string mFont;
+    /**
+     * When the font is specified either via the constructor or \ref setFont,
+     * this is set to ``true`` to indicate that \ref setTheme should not
+     * overwrite the user-specified default.  Otherwise, when \ref setTheme is
+     * called, the theme's default window font will be used.
+     */
+    bool mFontExplicit = false;
+    /**
+     * There are two default fonts in NanoGUI: ``"sans"`` and ``"sans-bold"``.
+     * When this variable is ``true``, the bold font will be used.  When
+     * ``false``, the regular font face will be used.  Subclasses are
+     * responsible for specifying this behavior when calling the parent class
+     * constructor (\ref Widget::Widget).  For example, \ref nanogui::Label will
+     * always set this to ``false``, whereas \ref nanogui::Button will always
+     * set this to ``true``.
+     */
+    bool mFontDefaultIsBold;
```

Comment:
The same here. I don't think it's a good tradeoff to include a top-level field just to indicate whether a widget wants a bold font by default. You could instead have ``setDefaultFonts`` be a virtual method that is overridden by (few) widgets.